core/src/main/java/org/geysermc/geyser/translator/collision/GlassPaneAndIronBarsCollision.java

```diff
@@ -0,0 +1,106 @@
+
+import lombok.EqualsAndHashCode;
+import org.geysermc.geyser.level.physics.BoundingBox;
+import org.geysermc.geyser.session.GeyserSession;
+
+@EqualsAndHashCode(callSuper = true)
+@CollisionRemapper(regex = "glass_pane$|iron_bars$", usesParams = true, passDefaultBoxes = true)
+public class GlassPaneAndIronBarsCollision extends BlockCollision {
+    /**
+     * 1 = north
+     * 2 = east
+     * 3 = south
+     * 4 = west
+     * 5 = north, east
+     * 6 = east, south
+     * 7 = south, west
+     * 8 = west, north
+     */
+    private int facing;
+
+    public GlassPaneAndIronBarsCollision(String params, BoundingBox[] defaultBoxes) {
+        super(defaultBoxes);
+        //east=true,north=true,south=true,west=true
+        if (params.contains("north=true") && params.contains("east=false") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 1;
+        } else if (params.contains("north=true") && params.contains("east=true") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 5;
+        } else if (params.contains("east=true") && params.contains("north=false") && params.contains("south=false") && params.contains("west=false")) {
+            facing = 2;
+        } else if (params.contains("east=true") && params.contains("north=false") && params.contains("south=true") && params.contains("west=false")) {
+            facing = 6;
+        } else if (params.contains("south=true") && params.contains("north=false") && params.contains("east=false") && params.contains("west=false")) {
+            facing = 3;
+        } else if (params.contains("south=true") && params.contains("north=false") && params.contains("east=false") && params.contains("west=true")) {
+            facing = 7;
+        } else if (params.contains("west=true") && params.contains("north=false") && params.contains("east=false") && params.contains("south=false")) {
+            facing = 4;
+        } else if (params.contains("west=true") && params.contains("north=true") && params.contains("east=false") && params.contains("south=false")) {
+            facing = 8;
+        }
+    }
+
+    @Override
+    public boolean correctPosition(GeyserSession session, int x, int y, int z, BoundingBox playerCollision) {
+        boolean result = super.correctPosition(session, x, y, z, playerCollision);
+        playerCollision.setSizeX(playerCollision.getSizeX() - 0.0001);
+        playerCollision.setSizeY(playerCollision.getSizeY() - 0.0001);
+        playerCollision.setSizeZ(playerCollision.getSizeZ() - 0.0001);
+
+        if (this.checkIntersection(x, y, z, playerCollision)) {
+            switch (facing) {
+                case 1 -> playerCollision.setMiddleZ(z + 0.8625); // North
+                case 2 -> playerCollision.setMiddleX(x + 0.1375); // East
+                case 3 -> playerCollision.setMiddleZ(z + 0.1375); // South
+                case 4 -> playerCollision.setMiddleX(x + 0.8625); // West
```

Comment:
I know its not quite the case with the other classes, but can we pull these out to named constants